blas/gonum/level2cmplx128.go

```diff
@@ -78,3 +79,102 @@ func (Implementation) Zgeru(m, n int, alpha complex128, x []complex128, incX int
 		kx += incX
 	}
 }
+
+// Zher performs the Hermitian rank-one operation
+//  A += alpha * x * x^H
+// where A is an n×n Hermitian matrix, alpha is a real scalar, and x is an n
+// element vector. On entry, the imaginary parts of the diagonal elements of A
+// are ignored and assumed to be zero, on return they will be set to zero.
+func (Implementation) Zher(uplo blas.Uplo, n int, alpha float64, x []complex128, incX int, a []complex128, lda int) {
+	if uplo != blas.Upper && uplo != blas.Lower {
+		panic(badUplo)
+	}
+	checkZMatrix('A', n, n, a, lda)
+	checkZVector('x', n, x, incX)
+
+	if n == 0 || alpha == 0 {
+		return
+	}
+
+	var kx int
+	if incX < 0 {
+		kx = (1 - n) * incX
+	}
+	if uplo == blas.Upper {
+		if incX == 1 {
+			for i := 0; i < n; i++ {
+				if x[i] != 0 {
+					tmp := complex(alpha*real(x[i]), alpha*imag(x[i]))
```

Comment:
To do `alpha*cmplx.Conj(x[i])` I would have to convert `alpha` to `complex128` which will be slower. Correct, I cannot do the conjugation in `tmp` because I'm operating on rows. However, now I'm thinking that simply operating on columns like in `Zgerc` would be simpler and probably faster: conjugation is done in tmp, and then asm Zaxpy updates the column. Do you agree?